app/models/metric/ci_mixin/capture.rb

```diff
@@ -190,7 +190,9 @@ def perf_capture_callback(task_ids, _status, _message, _result)
           # Task is done, call the rollup on the parent
           task.state, task.status, task.message = [MiqTask::STATE_FINISHED, MiqTask::STATUS_OK, "Performance collection complete, #{task.context_data[:complete].length} out of #{task.context_data[:targets].length} collections completed"]
 
-          pclass, pid = task.context_data[:parent].split(":")
+          # Splitting e.g. "ManageIQ::Providers::Openstack::InfraManager::EmsCluster:8" to class and id, we split by
+          # ':', using negative lookbehind and lookahead to check there is not another ':' before or after the one ':'.
+          pclass, pid = task.context_data[:parent].split(/(?<!:):(?!:)/)
```

Comment:
This seems overly complicated.  The last `:` should always be the divider, so would this work?

```ruby
pclass, _, pid = task.context_data[:parent].rpartition(":")
```